docs/reference/upgrade/rolling_upgrade.asciidoc

```diff
@@ -13,12 +13,16 @@ into the following two groups and upgrade the groups in this order:
 . Nodes that are not <<master-node,master-eligible>>. You can retrieve a list
 of these nodes with `GET /_nodes/_all,master:false` or by finding all the nodes
 configured with `node.master: false`.
+.. If you are using data tiers, you should
+upgrade the nodes by tier starting with frozen, cold, warm then hot-content tier.
+This is to ensure ILM can continue to move between the phases and ensure version
+compatibility.
+.. If you are not using data tiers, you may upgrade the nodes within the group in any order.
```

Comment:
I think we should ideally call out that if "you are using node attributes for hot/warm deployment, upgrade the colder tier first."